src/python/aqueduct_executor/operators/connectors/tabular/relational.py

```diff
@@ -24,7 +25,12 @@ def discover(self) -> List[str]:
         return inspect(self.engine).get_table_names()
 
     def extract(self, params: extract.RelationalParams) -> pd.DataFrame:
-        df = pd.read_sql(params.query, con=self.engine)
+        query = params.query
+        if "{{today}}" in query:
```

Comment:
I think variation like {{ today }} and {{today }} should also be valid. Let's look for whats in between {{ and }} and strip out the whitespace. Then we take all the arguments extracted in this fashion and try to match it against a predefined map from [arg name] to func() -> str which replaces that string.

This will let us extend this easily to other sql arguments besides today.

Reply:
There might be better way of doing this, but you can just keep track of the indexes in the string that the argument starts and ends at, eg: [(5, 10), (15, 20), (36, 55)], then perform the string replacements in reverse.

The point is that we will want to support in the future 1) other types of sql arguments and 2) multiple sql arguments in a single statement

Reply:
Yeah, I think that'll be ideal. I am trying to implement a version using RegEx so it extracts all the curly brace.